vxquery-core/src/main/java/org/apache/vxquery/serializer/XMLSerializer.java

```diff
@@ -346,6 +358,100 @@ private void printAttributeNode(PrintStream ps, TaggedValuePointable tvp) {
         }
     }
 
+    private void printObject(PrintStream ps, TaggedValuePointable tvp) {
+        ObjectPointable op = pp.takeOne(ObjectPointable.class);
+        TaggedValuePointable keys = pp.takeOne(TaggedValuePointable.class);
+        tvp.getValue(op);
+        try {
+            op.getKeys(keys);
+            printPair(ps, keys, op);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            pp.giveBack(keys);
+        }
+    }
+
+    private void printPair(PrintStream ps, TaggedValuePointable keys, ObjectPointable op) {
+        SequencePointable seqp = pp.takeOne(SequencePointable.class);
```

Comment:
Split this function into two functions. One to print all specified keys from object argument and one to print a single key from the object argument.
Example:
 - private void printObjectPairs(PrintStream ps, TaggedValuePointable keys, ObjectPointable op);
 - private void printObjectPair(PrintStream ps, TaggedValuePointable key, ObjectPointable op)